src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Controllers/PublicationsControllerTests.cs

```diff
@@ -419,6 +481,76 @@ public ListDataSetsTests(TestApplicationFactory testApp) : base(testApp)
             Assert.All(content.Results, r => Assert.Contains(expectedDataSetIds, id => id == r.Id));
         }
 
+        [Fact]
+        public async Task ReturnsCorrectViewModel()
+        {
+            var publicationId = Guid.NewGuid();
+
+            var dataSet = DataFixture
+                .DefaultDataSet()
+                .WithStatusPublished()
+                .WithPublicationId(publicationId)
+                .Generate();
+
+            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+
+            var dataSetVersion = DataFixture
+                .DefaultDataSetVersion(
+                    filters: 1,
+                    indicators: 1,
+                    locations: 1,
+                    timePeriods: 3)
+                .WithStatusPublished()
+                .WithDataSet(dataSet)
+                .FinishWith(dsv => dataSet.LatestVersion = dsv)
+                .Generate();
+
+            await TestApp.AddTestData<PublicDataDbContext>(context =>
+            {
+                context.DataSetVersions.Add(dataSetVersion);
+                context.DataSets.Update(dataSet);
+            });
+
+            var client = BuildApp().CreateClient();
+
+            var response = await ListDataSets(
+                client: client,
+                publicationId: publicationId,
+                page: 1,
+                pageSize: 1);
+
+            var content = response.AssertOk<DataSetPaginatedListViewModel>(useSystemJson: true);
+
+            Assert.NotNull(content);
+            Assert.Equal(1, content.Paging.Page);
+            Assert.Equal(1, content.Paging.PageSize);
+            Assert.Equal(1, content.Paging.TotalResults);
+
+            var result = Assert.Single(content.Results);
+
+            Assert.Equal(dataSet.Id, result.Id);
+            Assert.Equal(dataSet.Title, result.Title);
+            Assert.Equal(dataSet.Summary, result.Summary);
+            Assert.Equal(dataSet.Status, result.Status);
+            Assert.Equal(dataSet.SupersedingDataSetId, result.SupersedingDataSetId);
+            Assert.Equal(dataSetVersion.Version(), result.LatestVersion.Number);
+            Assert.True(Math.Abs(dataSetVersion.Published!.Value.Subtract(result.LatestVersion.Published).TotalMilliseconds) < 1);
```

Comment:
Only checking for millisecond precision here because Postre only stores the dates to millisecond precision. Whereas, `DateTimeOffset` stores to 100ns precision. So this is a flakey test unless I remove the 7th digit